include/openthread/unstable/link.h

```diff
@@ -0,0 +1,83 @@
+/*
+ *  Copyright (c) 2025, The OpenThread Authors.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ * @brief
+ *  This file defines the OpenThread unstable IEEE 802.15.4 Link Layer API.
+ */
+#ifndef OPENTHREAD_UNSTABLE_LINK_H_
+#define OPENTHREAD_UNSTABLE_LINK_H_
+
+#include <openthread/link.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+/**
+ * @addtogroup unstable-api-link
+ *
+ * @brief
+ *   This module includes unstable functions that control link-layer configuration.
+ *
+ * @{
+ */
+
+/**
+ * Defines constants of wake-up address types.
+ */
+typedef enum otWakeupAddressType
+{
+    OT_WAKEUP_ADDRESS_TYPE_WAKEUP_ID,       ///< Wake-up Identifier.
+    OT_WAKEUP_ADDRESS_TYPE_GROUP_WAKEUP_ID, ///< Group wake-up Identifier.
+    OT_WAKEUP_ADDRESS_TYPE_EXT_ADDRESS,     ///< Extended Address.
+} otWakeupAddressType;
+
+/**
+ * Represents whether to use Wake-up Identifier or Extended Address to wake up the Wake-up Listener.
+ */
+typedef struct otWakeupAddress
+{
+    union
+    {
+        uint64_t     mWakeupId;   ///< Wake-up identifier of the Wake-up Listener.
+        otExtAddress mExtAddress; ///< IEEE 802.15.4 Extended Address of the Wake-up Listener.
+    } mShared;
+
+    otWakeupAddressType mType : 2; ///< Indicates the wake-up address type (see `OT_WAKEUP_ADDRESS_TYPE_*`).
```

Comment:
```suggestion
    otWakeupBy mWakeupBy : 2; ///< Indicates the wake-up type (see `OT_WAKEUP_BY_*`).
```

Reply:
How about changing it to `otWakeupType`?